File: web/lib/signal-labels.ts
/** Presentation only: keep persisted values and Python decisions intact. */
export const trendExplanation = "Trend Adayı, teknik trend koşullarını karşılayan varlığı gösterir. Alım zamanlamasını merkezi karar belirler; şirketin finansal sağlamlığı bu etiketle değerlendirilmez.";
export const confidenceExplanation = "Algoritma güven puanı teknik uyumu özetler; ölçülmüş başarı olasılığı değildir.";

export function technicalProfile(value: unknown): string {
  return String(value ?? "—").replace(/uzun vadel[iİı] aday/giu, "TREND ADAYI") || "—";
}

export function confidenceScore(value: unknown): string {
  if (value === null || value === undefined || typeof value === "boolean") return "—";
  const raw = String(value).trim().replaceAll("%", "").replace(/\/100$/, "").trim().replace(",", ".");
  if (!/^\d+(?:\.\d+)?$/.test(raw)) return "—";
  const number = Number(raw);
  return Number.isFinite(number) && number >= 0 && number <= 100 ? `${number}/100` : "—";
}

export function isTrendCandidate(row: Record<string, unknown>): boolean {
  return technicalProfile(row["Teknik Profil"]).toLocaleUpperCase("tr-TR").includes("TREND ADAYI");
}

File: izfin_ui/signal_labels.py
"""Display-only compatibility for stored technical profiles and confidence."""
from __future__ import annotations

import math
import re

TREND_ACIKLAMASI = (
    "Trend Adayı, teknik trend koşullarını karşılayan varlığı gösterir. "
    "Alım zamanlamasını merkezi karar belirler; şirketin finansal sağlamlığı "
    "bu etiketle değerlendirilmez."
)
GUVEN_ACIKLAMASI = "Algoritma güven puanı teknik uyumu özetler; ölçülmüş başarı olasılığı değildir."


def teknik_profil_etiketi(value) -> str:
    text = str(value or "—")
    return re.sub(r"UZUN VADELİ ADAY", "TREND ADAYI", text, flags=re.IGNORECASE)


def trend_adayi_mi(value) -> bool:
    return "TREND ADAYI" in teknik_profil_etiketi(value).upper()


def karar_metni_etiketi(value) -> str:
    """Keep stored rationale intact while displaying confidence as a score."""
    return re.sub(r"algoritma güveni (yüksek|sınırlı) \(%(\d+)\)",
                  r"algoritma güven puanı \1 (\2/100)", str(value or ""), flags=re.IGNORECASE)


def guven_puani_etiketi(value) -> str:
    if value is None or isinstance(value, bool):
        return "—"
    text = str(value).strip().replace("%", "").removesuffix("/100").strip()
    try:
        number = float(text.replace(",", "."))
    except (ValueError, TypeError):
        return "—"
    return f"{number:g}/100" if math.isfinite(number) and 0 <= number <= 100 else "—"
